starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -78,13 +80,10 @@ public Optional<Inventory> update(String id, Inventory inventory) {
   /**
    * Delete Inventory By Id.
    * @param id Id of Inventory.
-   * @return Deleted Inventory.
    */
-  public Optional<Inventory> delete(String id) {
-    Inventory inventory = mongoTemplate.findById(id, Inventory.class);
-    if (inventory != null) {
-      mongoTemplate.remove(inventory);
-    }
-    return Optional.of(inventory);
+  public void delete(String id) {
+    
+    mongoTemplate.findAndRemove(new Query().addCriteria(Criteria.where("id").is(id)), Inventory.class);
+    
```

Comment:
We do still want to have the return here and we can leave the optional so that if the query does not return something, we can encapsulate that possibility. Take a look at the Optional API for a method that can help us with this.

Reply:
The query here is good!